src/adapters/scene.js

```diff
@@ -145,10 +147,29 @@ Scene.prototype.initMapBox = function() {
       }, this.DOUBLE_TAP_DELAY_MS);
     });
 
+    // Long touch on mobile (500ms)
+    // Custom implementation because the contextmenu event isn't supported by MapBox.
+    // Long touch is initiated with a touchstart event
+    // It is canceled if many fingers touch the map, or if one of these events is initiated:
+    // touchmove, touchcancel, touchend, pointerdrag, pointermove, moveend, gesturestart, gesturechange, gestureend
     if (isMobileDevice()) {
-      this.mb.on('contextmenu', e => {
-        this.clickOnMap(e.lngLat, null, { longTouch: true });
+      this.mb.on('touchstart', e => {
+        if (e.originalEvent.touches.length === 1) {
+          longTouchTimeout = setTimeout(() => {
+            this.clickOnMap(e.lngLat, null, { longTouch: true });
+          }, 500);
+        }
       });
+
+      this.mb.on('touchend', this.cancelLongTouch);
```

Comment:
Instead of the big redundancy in these instructions and comments, it would be simpler to declare an array with a good name and loop.

```js
const longTouchCancellingEvents = [ 'touchend', 'touchcancel', 'touchmove' /*…*/ ];
const cancelLongTouch = () => { clearTimeout(longTouchTimeout); }
longTouchCancellingEvents.forEach(event => { this.mb.on(event, cancelLongTouch);  });
```

That way everything related to cancellation is grouped, you don't need the class method and you don't need the redundant comment with the event list, as the code is self-explanatory.

Reply:
I thought about doing it initially, but decided against it as it seemed a bit less simple to read in this form.
However, I agree to do the change you propose, it's fine too. :)